comictaggerlib/ui/customwidgets.py

```diff
@@ -112,3 +122,256 @@ def toggleItem(self, index: int) -> None:
             self.setItemChecked(index, False)
         else:
             self.setItemChecked(index, True)
+
+
+# Inspiration from https://github.com/marcel-goldschen-ohm/ModelViewPyQt and https://github.com/zxt50330/qitemdelegate-example
+class ReadStyleItemDelegate(QtWidgets.QStyledItemDelegate):
+    buttonClicked = pyqtSignal(QModelIndex, ClickedButtonEnum)
+
+    def __init__(self, parent: QtWidgets.QWidget):
+        super().__init__()
+        self.combobox = parent
+
+        self.button_width = 24
+        self.button_padding = 5
+
+        # Connect the signal to a slot in the delegate
+        self.combobox.itemClicked.connect(self.itemClicked)
+
+    def paint(self, painter: QtGui.QPainter, option: QtWidgets.QStyleOptionViewItem, index: QModelIndex) -> None:
+        options = QtWidgets.QStyleOptionViewItem(option)
+        self.initStyleOption(options, index)
+        style = self.combobox.style()
+
+        # Draw background with the same color as other widgets
+        palette = self.combobox.palette()
+        background_color = palette.color(QtGui.QPalette.Window)
+        painter.fillRect(option.rect, background_color)
+
+        style.drawPrimitive(QtWidgets.QStyle.PE_PanelItemViewItem, options, painter, self.combobox)
+
+        painter.save()
+
+        # Checkbox drawing logic
+        checked = index.data(Qt.CheckStateRole)
+        opts = QtWidgets.QStyleOptionButton()
+        opts.state |= QtWidgets.QStyle.State_Active
+        if index.flags() & Qt.ItemIsEditable:
+            opts.state |= QtWidgets.QStyle.State_Enabled
+        else:
+            opts.state |= QtWidgets.QStyle.State_ReadOnly
+        if checked:
+            opts.state |= QtWidgets.QStyle.State_On
+        else:
+            opts.state |= QtWidgets.QStyle.State_Off
+        opts.rect = self.getCheckBoxRect(option)
+        style.drawControl(QtWidgets.QStyle.CE_CheckBox, opts, painter)
+
+        label = index.data(Qt.DisplayRole)
+        rectangle = option.rect
+        rectangle.setX(opts.rect.width() + 5)
+        painter.drawText(rectangle, Qt.AlignVCenter, label)
+
+        # Draw buttons
+        if checked and (option.state & QtWidgets.QStyle.State_Selected):
+            # TODO Fill button rects or similar for better separation from text below?
+            up_rect = self._button_up_rect(option.rect)
+            down_rect = self._button_down_rect(option.rect)
+
+            painter.drawImage(up_rect, QtGui.QImage(str(graphics_path / "up.png")))
+            painter.drawImage(down_rect, QtGui.QImage(str(graphics_path / "down.png")))
```

Comment:
At the moment the icon arrows can get a bit lost with any text underneath. Adding a little extra length to the `sizeHint` or `setMinimumContentsLength` might be a way around it.

Reply:
You should just be able to add up the width of all the widgets and give it in the sizeHint